### ISSUE TITLE
[13.0] mail: wrong sql in  _get_recipient_data

### ISSUE THREAD

**Issue by tanghulu0608** [odoo/odoo#47179]
https://github.com/odoo/odoo/blob/af411b866aa2052c01168342f8f05ae3dabad83e/addons/mail/models/mail_followers.py#L171

when pids and cids have some values, then got a sql like this:

```
SELECT DISTINCT ON (partner.id) partner.id as pid, NULL::int AS cid,
    partner.active as active, partner.partner_share as pshare, NULL as ctype,
    users.notification_type AS notif, NULL AS groups
FROM res_partner partner
LEFT JOIN res_users users ON users.partner_id = partner.id AND users.active
WHERE partner.id IN (3,10)
ORDER BY partner.id, users.notification_type 
UNION
SELECT NULL::int AS pid, channel.id AS cid,
    TRUE as active, NULL AS pshare, channel.channel_type AS ctype,
    CASE when channel.email_send = TRUE then 'email' else 'inbox' end AS notif, NULL AS groups
FROM mail_channel channel WHERE channel.id IN  (4) 
```

this will raise a err in postgresql :
```

ERROR:  syntax error at or near "UNION"
LINE 9: UNION
        ^
SQL state: 42601
Character: 376
```

**Comment by pedrobaeza** [comment]
Do you have any step to reproduce the problem via UI?

**Comment by tanghulu0608** [comment]
Sorry,I can not reproduce the problem via UI.
`           self.message_notify(
                subject=_('Relaunch Error'),
                body=_('<p>Substrate insufficient for %s </p>') % self.name,
                partner_ids=group.users.mapped("partner_id").ids,
                channel_ids=channel.ids,
            )
`
This is my code. when partner_ids and channel.ids have value, get the error.

**Comment by pedrobaeza** [comment]
I think you have to make 2 calls, each one with the subscription, as this is not supported. @tde-banana-odoo can you confirm?

**Comment by tanghulu0608** [comment]
but the method _get_recipient_data seems to support both  partner_ids and channel.ids

**Comment by tde-banana-odoo** [comment]
I don't think it is intended that giving both partner_ids and channel_ids fails. It is not a common use case of notify though :) . Issue seem therefore valid; could be great to have a failing tests in test_mail_follower.py file ^^ .

**Comment by C3POdoo** [comment]
Dear @tanghulu0608,

Thank you for your report but we are closing it due to inactivity.
We apology if we could not look at your request in time.
If your report still makes sense, don't hesitate to reopen a new one. We will try to check it as soon as possible.

_This is an automated message._
